tensorflow_datasets/image/the300w_lp.py

```diff
@@ -0,0 +1,149 @@
+            "tex_params": tfds.features.Tensor(shape=(199,), dtype=tf.float32),
+            "shape_params": tfds.features.Tensor(shape=(199,),
+                                                 dtype=tf.float32),
+            "exp_params": tfds.features.Tensor(shape=(29,), dtype=tf.float32),
+            "pose_params": tfds.features.Tensor(shape=(7,), dtype=tf.float32)
+        }),
+        urls=[_PROJECT_URL],
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    extracted_path = dl_manager.download_and_extract(_DATASET_URL)
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=1,
+            gen_kwargs={
+                "image_dir_path": os.path.join(extracted_path, '300W_LP'),
+            }),
+    ]
+
+  def _generate_examples(self, image_dir_path):
+    """Yields examples."""
+    image_files = tf.io.gfile.glob(
+        pattern=os.path.join(image_dir_path, "[!Code]*[!_Flip]/[!_]*.jpg"))
+    label_files = list(map(lambda s: s.replace("jpg", "mat"), image_files))
+    landmark_files = map(lambda s: s.replace("300W_LP", "300W_LP/landmarks"),
+                         image_files)
+    landmark_files = list(map(lambda s: s.replace(".jpg", "_pts.mat"),
+                              landmark_files))
+    for image_file, label_file, landmark_file in zip(image_files,
+                                                     label_files,
+                                                     landmark_files):
+      with tf.io.gfile.GFile(label_file, "rb") as f:
+        mat = tfds.core.lazy_imports.scipy.io.loadmat(f)
+      pt2d_origin = mat["pt2d"].T
+      pt2d_origin = (pt2d_origin / 450.0).astype(np.float32)
+      roi = mat["roi"].reshape(4).astype(np.float32)
+      illum_params = mat["Illum_Para"].reshape([-1]).astype(np.float32)
+      color_params = mat["Color_Para"].reshape([-1]).astype(np.float32)
+      tex_params = mat["Tex_Para"].reshape([-1]).astype(np.float32)
+      shape_params = mat["Shape_Para"].reshape([-1]).astype(np.float32)
+      exp_params = mat["Exp_Para"].reshape([-1]).astype(np.float32)
+      pose_params = mat["Pose_Para"].reshape([-1]).astype(np.float32)
+      with tf.io.gfile.GFile(landmark_file, "rb") as f:
+        ldm_mat = tfds.core.lazy_imports.scipy.io.loadmat(f)
+        pt2d = (ldm_mat["pts_2d"] / 450.0).astype(np.float32)
+        pt3d = (ldm_mat["pts_3d"] / 450.0).astype(np.float32)
+      record = {
+          "image": image_file,
+          "landmarks_origin": pt2d_origin,
+          "landmarks_2d": pt2d,
+          "landmarks_3d": pt3d,
+          "roi": roi,
+          "illum_params": illum_params,
+          "color_params": color_params,
+          "tex_params": tex_params,
+          "shape_params": shape_params,
+          "exp_params": exp_params,
+          "pose_params": pose_params
+      }
+      yield image_file, record
```

Comment:
The key (in this case image_file) should not vary from one machine to the next.
Here the path will depend on where the user is running.

You should just use the basename (+ directory name if that is not enough to get uniqueness).

Reply:
> The key (in this case image_file) should not vary from one machine to the next.
> Here the path will depend on where the user is running.
> 
> You should just use the basename (+ directory name if that is not enough to get uniqueness).

changed to use basename as key in d9e394d, since no name conflicts in image filenames.